merge_state_block.py

```diff
@@ -1,50 +1,37 @@
-from .state_change_base_block import StateChangeBase, NoState
+from .state_base_block import StateBase
 from nio.common.discovery import Discoverable, DiscoverableType
-from nio.metadata.properties.expression import ExpressionProperty
-from nio.metadata.properties.bool import BoolProperty
-from nio.metadata.properties.string import StringProperty
+from nio.metadata.properties import ExpressionProperty
+from nio.metadata.properties import StringProperty
 
 
 @Discoverable(DiscoverableType.block)
-class MergeState(StateChangeBase):
-    '''
+class MergeState(StateBase):
+
+    """
     If *state_sig* evaluates to True then the signal sets the state according
     to *state_expr*. Else, the signal gets assigned the state to the attribute
     *state_name*. If no signal has been set, then *state_name* is set to None.
 
-    '''
+    """
 
     state_name = StringProperty(default='state', title="State Name")
     state_sig = ExpressionProperty(title="Is State Signal",
                                    default="{{hasattr($, 'state')}}")
-    use_persistence = BoolProperty(title="Use Persistence", default=False,
-            visible=False)
 
-    def process_signals(self, signals):
-        signal_list = []
+    def _process_group(self, signals, group, to_notify=list()):
         for signal in signals:
             try:
                 is_state_sig = self.state_sig(signal)
-            except Exception as e:
-                is_state_sig = False
-                self._logger.error(
-                    "Failed determining state signal: {}".format(e)
-                )
+            except:
+                self._logger.exception("Failed determining state signal")
```

Comment:
If the state_sig expression fails, what is the value of is_state_sig? Or should this return?